### ISSUE TITLE
Video and animation previews are not working for the most sites (bug or not?)

### ISSUE THREAD

**Issue by Dean-Corso** [hoothin/UserScripts#132]
Hi hoothin,

I like your script and it seems to work pretty well on many sites but I found one main problem when sites using thumbnail previews or animations etc. Mostly they are not working when your script did loaded the next pages below. If I move the mouse on a thumbnail then the preview does not start (mp4, webm or gif etc) and nothing happens. Just would like to know whether its a bug or whether I need to customize a own rule for all this site and if yes what is important to make the previews working?

I have test few websites using such preview thumbs (NSFW) and all of them are not working from page 2 and go on. So this is really bad of course. Below few sites you can check.

```
Not working thumbnail preview from page 2
----------------------------------------------
spankbang.com/new_videos/
xhamster.com
porntrex.com/latest-updates/

Working thumbnail preview from page 2
----------------------------------------------
pornhub.com/video?page=1
```
On all sites your script is working but the thumbnail / previews are not starting (mouse over) anymore from page 2. The only exception is the pornhub site. So what I have to do to make it work for all? Thank you.

PS: I also tried a AddOn & Extension called "uAutoPagerize" with the same bad results (previews not working). Is it a bug in both (extension and your script) or is it just not working at all?

**Comment by hoothin** [comment]
Yes, it's because the first priority of pagetual is load next page with static content, any dynamic content should loaded with "action: 1" or "action: 2" but much more slowly.
You can try menu command "Force to join". Then every content on current site will be loaded in dynamic mode.

**Comment by Dean-Corso** [comment]
Hello hoothin.

thanks for your answer. I tried using the action parameter 2 and it works for that website.
```
[
    {
        "name": "SpankBang",
        "url": "^https://(.*\\.)?spankbang\\.com/",
        "pageElement": "BODY > MAIN:nth-of-type(1) > DIV.listing_page > DIV.main_results > DIV.results.results_search > DIV.video-list.video-rotate.video-list-with-ads>*",
        "action": 2
    }
]
```
I removed that rule again because your script does already load the site pages of that website and I tried to enable that "Force to join" option on this website and now it seems to work. Of course its not looking so nice anymore but the previews are working. 

I have another question about using the "action 2" parameter. I tried some sites now just adding this command and it really seems to work for all I have test but in some cases the webpage is getting smaller if on the left side is a panel to see. Just go on the mainpage of xhamster and you see on the left side the panel you can pick any category etc. Now if I set a rule for xhamster with action 2 then the pages starting from page 2 are smaller and everything is moved inside of the "index-videos.mixed-section" element (category & video thumbnails) and on left side is free space. What parameter I have to use in this case to show it correctly (video thumbs) as on page 1? Thank you.

**Comment by Dean-Corso** [comment]
Hi hoothin,

so when using just "body" as pageElement then its looking not so good so I would like to make some better looking custom rules in such cases but its tricky to understand all of the rules parameters I can / could use.

Problems: Some sites are just loading few pages and then your script loads empty pages and I just can see your page bar with the page number 4,5,6 etc but there showing no content only the page bars. In some other cases the thumbnails are not loading and keep empty or they are showing one by one below instead of side by side etc. I found different situations I need to handle manually and create rules but its not easy so far. So at the moment I just can use the basic parameters of url, pageElement, nextLink & action which I can find manually by using inspect tool of browser to find right elements/classes etc but what about the rest? Do you have maybe some more detailed examples I can check out?

Example: startpage.com is not working. This site is listed in your pagetualRules.json file...
[startpage](https://github.com/hoothin/UserScripts/blob/master/Pagetual/pagetualRules.json#L69)
...so its trying to load to show more results but it fails. The rules for nextLinkByJs parameter is pretty long and complex. Why? I tried to create a simple rule like this...
```
    {
        "name": "Startpage",
        "url": "^https://www\\.startpage\\.com/",
        "pageElement": "section.w-gl w-gl--desktop w-gl--",
        "nextLinkByJs": "button.pagination__next-prev-button next"
    }
```
...but failed too. Do you have made some tutorials or anything I could check out to get more advanced dealing with your script & rule commands? Thank you.

**Comment by hoothin** [comment]
![image](https://user-images.githubusercontent.com/5361213/178861166-76594c8e-1251-468c-9fc5-3806160a7d38.png)
I test Startpage with the right result👆.
The "nextLinkByJs" is used for writing javascript code to get element of next link. You can learn javascript from [w3schools](https://www.w3schools.com/js/default.asp).
I think you enter the incorrect css selector of pageElement so you can just see your page bar with the page number.
About the case you found bad looking, best way is to set with "body", if you don't want to write custom css for that site.

**Comment by Dean-Corso** [comment]
Hi again,

about Startpage. I'am using latest Firefox & Brave brwoser (x64 / Win 10) and in both your script does fail to load Startpage / pages and in console window I see same debug infos logged.

```
Firefox
BODY > DIV.layout > DIV.layout-web.layout-web-- > DIV.layout-web__body.layout-web__body--desktop > DIV.layout-web__mainline > DIV.show-results > DIV.mainline-results-- > DIV.mainline-results.mainline-results__web > SECTION.w-gl.w-gl--desktop.w-gl-->*

Brave
BODY > DIV.layout > DIV.layout-web.layout-web-- > DIV.layout-web__body.layout-web__body--desktop > DIV.layout-web__mainline > DIV.show-results>*

Both
Stop as no page when emu
```
Ok, I will make some tests with the other sites to find out how to make it right. If I don't get it work I will ask you later again to get some working rule from you in best case and to lern a little. Thank you.

**Comment by hoothin** [comment]
It looks like that script run without any rule in your case. I think you should import the base rule on config page at first.

**Comment by Dean-Corso** [comment]
Hi,

the basic rule is added. Under the update button on your Pagetual site I can see 2 times this URL..
`http://wedata.net/databases/AutoPagerize/items_all.json`
...with arrow up & down & x to remove it. I did already added the base rule.
```
http://wedata.net/databases/AutoPagerize/items_all.json
https://github.com/hoothin/UserScripts/blob/master/Pagetual/pagetualRules.json
```
This URLs I can see on my Pagetual settings page but startpage is not working for me. Its just trying to load at the bottom but now new pages are comming and I need to press the next page button manually. How to make it work now?

**Comment by hoothin** [comment]
You need to delete that duplicate wedata rule and import `https://raw.githubusercontent.com/hoothin/UserScripts/master/Pagetual/pagetualRules.json` instead of `https://github.com/hoothin/UserScripts/blob/master/Pagetual/pagetualRules.json`

**Comment by Dean-Corso** [comment]
Somehow it dosen't work. Now I have 2 times the "http://wedata.net/databases/AutoPagerize/items_all.json" listed but I can't remove them (can't delete buildin rules) and when I try to add "0 | https://raw.githubusercontent.com/hoothin/UserScripts/master/Pagetual/pagetualRules.json" and save and the site does refresh then this JSON URL was not added and I see still 2 times the "http://wedata.net/databases/AutoPagerize/items_all.json". Maybe you could over work this issue and make it easier or better to add the basic URLs. Below a image...
![P_2022-07-16_231145](https://user-images.githubusercontent.com/88513086/179372060-646886a1-5b5b-46fa-94f1-b067293c7e92.png)
...not working to remove or add. What do I wrong? Can you not add the rules JSON URL automatically too?

**Comment by hoothin** [comment]
Yeah! A bug! I'll fix it soon.
You can simply import base rule by one click the rule line(shown when the script is initialization).
You triggered this error because you add "0|" when the rule is not on autopagerize format.

**Comment by Dean-Corso** [comment]
Hi,

yes I do click on the rule line and then it does update the wdata. Ok so now I tried to add this line again.."0 | https://raw.githubusercontent.com/hoothin/UserScripts/master/Pagetual/pagetualRules.json" and now it does add it and I can see 3 URL links. So two times the wedata URL (still can not remove one of them) and the third URL to pagetualRules.json URL. I tried to check out startpage and now it works. 😃 Also in both browser FF & Brave (updated script manually). Very nice and thank you.
